contracts/renderers/MetadataRegistry.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.17;
+
+import "../globals/Globals.sol";
+import "../crowdfund/Crowdfund.sol";
+import "../party/Party.sol";
+
+struct ContractMetadata {
+    string name;
+    string description;
+    string image;
+    string banner;
+}
+
+struct TokenMetadata {
+    string name;
+    string description;
+    string image;
+}
+
+contract MetadataRegistry {
+    error MetadataAlreadySetError();
+    error OnlyAllowedHostsError();
+    error OnlyMultisigError();
+    error InvalidCrowdfundError();
+
+    Globals private immutable _GLOBALS;
+
+    /// @notice The address of the host that is allowed to set custom token URIs
+    mapping(address => bool) public allowedHosts;
+    /// @notice The custom token URI metadata for the party created by a given crowdfund
+    mapping(Crowdfund => TokenMetadata) public customPartyTokenURIByCrowdfund;
+    /// @notice The custom contract URI metadata for the party created by a given crowdfund
+    mapping(Crowdfund => ContractMetadata) public customPartyContractURIByCrowdfund;
+
+    constructor(Globals global, address[] memory _allowedHosts) {
+        _GLOBALS = global;
+
+        for (uint256 i; i < _allowedHosts.length; i++) {
+            allowedHosts[_allowedHosts[i]] = true;
+        }
+    }
+
+    modifier onlyAllowedHosts() {
```

Comment:
could consider allowing multisig to allow anyone to set custom metadata, but this is optional